presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/sketch/SecureRandomizationStrategy.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation.sketch;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import java.security.SecureRandom;
+
+/**
+ * Note: Due to finite-precision implementation details, usage of floating-point functions
+ * as random noise, while cryptographically secure, may leak information from a privacy context.
+ * See "On Significance of the Least Significant Bits For Differential Privacy" by Mironov
+ * and use judiciously.
+ */
+public class SecureRandomizationStrategy
+        extends RandomizationStrategy
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(SecureRandomizationStrategy.class).instanceSize();
+    private static final SecureRandom random = new SecureRandom();
```

Comment:
Per https://arxiv.org/pdf/2002.04049.pdf, we should use a cryptographically secure PRNG. I was under the impression that `SecureRandom` would essentially use a secure seed, then deterministically generate from a PRNG, though on closer inspection, that appears to depend on the exact implementation being used. Since I'm not an expert here, I'm open to suggestions on how best to proceed here to balance security and performance.

Reply:
`NativePRNGNonBlocking` is used for secure_random UDF in Presto: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java#L102

Here's a description on how it operates: https://www.synopsys.com/blogs/software-security/proper-use-of-javas-securerandom.html

In short it initializes the seed from `/dev/urandom` what doesn't block if the OS experiencing lack of entropy. The algorithm is based on SHA1PRNG which is considered cryptographically secure.

From what i understand the paper states that:

> “Random numbers” should be generated by a cryptographically secure pseudo-random number generator (CSPRNG).

It looks like pseudo-random should be fine, as long as it is cryptographically secure.